src/discovery/azure/Akka.Discovery.Azure/ClusterMemberTableClient.cs

```diff
@@ -0,0 +1,204 @@
+
+        public ClusterMemberTableClient(string serviceName, string connectionString, string tableName, ILoggingAdapter log)
+        {
+            _log = log;
+            _serviceName = serviceName;
+            _client = new TableClient(connectionString, tableName);
+        }
+
+        internal async ValueTask<ClusterMember> GetOrCreateAsync(
+            IPAddress address,
+            int port,
+            CancellationToken token = default)
+        {
+            if (_entity != null)
+                return _entity;
+
+            if (!await EnsureInitializedAsync(token))
+                return null;
+
+            var rowKey = ClusterMember.CreateRowKey(address, port);
+            var entry = await GetEntityAsync(rowKey, token);
+            if (entry != null)
+            {
+                _entity = entry;
+                if(_log.IsDebugEnabled)
+                    _log.Debug($"[{_serviceName}@{_entity.Address}:{_entity.Port}] Found existing entry row. " +
+                               $"Created: [{_entity.Created}], last update: [{_entity.LastUpdate}]");
+                await UpdateAsync(token);
+                return _entity;
+            }
+
+            var entity = ClusterMember.CreateEntity(_serviceName, address, port);
+            var response = await _client.AddEntityAsync(entity, token);
+            if (response.IsError)
+            {
+                _log.Error($"[{_serviceName}@{address}:{port}] Failed to insert entry row. " +
+                           $"Reason: {response.ReasonPhrase}");
+                return null;
+            }
+
+            _entity = ClusterMember.FromEntity(entity);
+            if(_log.IsDebugEnabled)
+                _log.Debug($"[{_serviceName}@{_entity.Address}{_entity.Port}] New entry row created.");
+            return _entity;
+        }
+
+        internal async Task<ImmutableList<ClusterMember>> GetAllAsync(long lastUpdate, CancellationToken token = default)
+        {
+            if (!await EnsureInitializedAsync(token))
+                return null;
+
+            var query = _client
+                .QueryAsync<TableEntity>($"PartitionKey eq '{_serviceName}' and {ClusterMember.LastUpdateName} ge {lastUpdate}")
+                .WithCancellation(token);
+
+            var list = ImmutableList.CreateBuilder<ClusterMember>();
+            await foreach (var entry in query)
+            {
+                list.Add(ClusterMember.FromEntity(entry));
+            }
+            
+            if(_log.IsDebugEnabled)
+                _log.Debug($"[{_serviceName}@{_entity.Address}:{_entity.Port}] Retrieved {list.Count} entry rows.");
+            return list.ToImmutableList();
+        }
+
+        internal async Task<bool> UpdateAsync(CancellationToken token = default)
+        {
+            if (!await EnsureInitializedAsync(token))
+                return false;
+
+            var original = _entity.LastUpdate;
+            _entity = _entity.Update();
```

Comment:
This updates the timestamp?

Reply:
Yes, the ClusterMember class is immutable, this creates a new ClusterMember instance with an updated LastUpdate field.